### PR TITLE
Swift 2.3 Compatibility

### PR DESCRIPTION
**Pull request by gblotter**
- Ran the Xcode 8 project conversion tool
- Needed to create a default case for the location authorization status and the location auth status manager switch statements.  For now I defaulted to `completion(.Authorized)` but am open to changing it

### CODE REVIEW

**Comment by MarkQSchultz** [comment]
In order for the CI stuff to pass, you'll need to edit the `.travis.yml` file to have `osx_image: xcode8` or whatever is appropriate.

**Comment by gblotter** [comment]
@MarkQSchultz I'm seeing: "Signing for 'PSOperationsTests' requires a development team.  For the PSOperationsTests target.  How do you want to handle that?

**Comment by MarkQSchultz** [comment]
@gblotter I believe Xcode 8 allows you to disable signing on targets. So just make sure it's not signed and I think that will take care of it.

**Comment by MarkQSchultz** [comment]
The last build failed with this:
```
Testing failed:
	“Use Legacy Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.
** TEST FAILED **
```

**Comment by gblotter** [comment]
@MarkQSchultz not sure what the deal is... I can run the tests locally in Xcode and they pass...

**Comment by gblotter** [comment]
@MarkQSchultz Can you run the build again?  The first time I switched on my system it failed... had issues launching the beta simulator.

**Comment by MarkQSchultz** [comment]
Restarted.

**Comment by MarkQSchultz** [comment]
And now we need to remove signing on PSOperationsAppForTests.

**Comment by gblotter** [comment]
See https://github.com/pluralsight/PSOperations/pull/68/commits/6364d4dd802e26653754531a59e6c8fe3077e5b6

**Comment by MarkQSchultz** [comment]
Hard to tell what failed this time, since the logs don't actually show a failure. But the last thing the logs show was happening was trying to sign `PSOperationsAppForTestsTests`

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 language: objective-c
-osx_image: xcode7.3
+osx_image: xcode8
 script: 
  - xcodebuild test -project PSOperations.xcodeproj -scheme PSOperations -destination 'platform=iOS Simulator,name=iPhone 6s' -destination 'platform=tvOS Simulator,name=Apple TV 1080p'
  - xcodebuild test -project PSOperations.xcodeproj -scheme PSOperationsAppForTests -destination 'platform=iOS Simulator,name=iPhone 6s'
```
Comment (regnerjr) [inline .travis.yml]: Suggest turning off code signing. Just append  ` CODE_SIGNING_REQUIRED=NO` to the end of you `xcodebuild` command.
Similar to this commit https://github.com/Electrode-iOS/ELReachability/commit/5ef915f53065bc585ec7f3c54e93e18f7dca9f79

**Pull request closed without merging**